cloudstack_plugin/virtual_machine.py

```diff
@@ -602,6 +606,61 @@ def disconnect_floating_ip(ctx, **kwargs):
     return True
 
 
+@operation
+def attach_volume(ctx, **kwargs):
+    """ Attach a volume to virtual machine.
+    """
+
+    cloud_driver = get_cloud_driver(ctx)
+
+    server_id = ctx.source.instance.runtime_properties[CLOUDSTACK_ID_PROPERTY]
+    volume_id = ctx.target.instance.runtime_properties[CLOUDSTACK_ID_PROPERTY]
+
+    server = get_vm_by_id(ctx, cloud_driver, server_id)
+    volume = get_volume_by_id(cloud_driver, volume_id)
+
+    if server is None:
+        raise NonRecoverableError('Server with id {0} not found'
+                                  .format(server_id))
+
+    if volume is None:
+        if volume is None:
+            raise NonRecoverableError('Volume with id {0} not found'
+                                      .format(volume_id))
+
+    cloud_driver.attach_volume(node=server,
+                               volume=volume)
```

Comment:
what about the device name? is it possible to set a runtime property on the source specifying the device name?